src/activity_handlers.cpp

```diff
@@ -1345,8 +1345,11 @@ void activity_handlers::train_finish( player_activity *act, player *p )
                                 new_skill_level, skill.name().c_str());
         }
 
-        // possible callback arguments: skill_increase_type, skill_id, skill_value_prev, skill_value_new
-        lua_callback( "on_skill_increased", "training" );
+        // possible callback arguments: skill_increase_type, skill_id, skill_level_new
+        const static auto skill_increase_type = "training";
+        const static auto skill_id = skill.ident().c_str();
+        const static auto skill_level_new = std::to_string( new_skill_level ).c_str();
```

Comment:
This creates a dangling pointer. `to_string` returns a value that only exists until the end of the expression. The pointer from `c_str` is invalid after that. Why don't you put all those values directly into the function call instead of having several variables that are used only once anyway?

Reply:
Thanks. That is just a stub. I'm trying to figure out convinient way of transferring callback arguments (which come in different type and numbers) to lua. I'm not sure yet what is the best approach.